src/workerd/api/sockets.h

```diff
@@ -69,27 +72,48 @@ struct SocketOptions {
 struct InitData {
   jsg::Ref<ReadableStream> readable;
   jsg::Ref<WritableStream> writable;
-  IoOwn<kj::PromiseFulfillerPair<void>> closeFulfiller;
+  IoOwn<kj::PromiseFulfillerPair<bool>> closeFulfiller;
+  IoOwn<kj::PromiseFulfillerPair<void>> jsCloseFulfiller;
+  kj::ForkedPromise<void> jsCloseFulfillerFork;
 };
 
 class Socket: public jsg::Object {
 public:
-  Socket(InitData data)
+  Socket(jsg::Lock& js, InitData data)
```

Comment:
Is this version of the constructor only meant to be called by the other overload of the constructor? If so, should it be private?